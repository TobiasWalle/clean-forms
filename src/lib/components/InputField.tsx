import * as React from 'react';
import { createField } from '../hocs';
import { getInputProps } from '../utils/getInputProps';

export const InputField = createField<string, JSX.IntrinsicElements['input']>(
  ({ input: fieldInputProps, custom }) => {
    return <input {...getInputProps(fieldInputProps, custom)} />;
  }
);
