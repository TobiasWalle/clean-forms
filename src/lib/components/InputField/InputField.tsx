import * as React from 'react';
import { createField } from '../../hocs';
import { FieldRenderFunction, FieldPropsWithoutRender } from '../Field/Field';

export type InputValue = number | string;
export type InputProps = React.InputHTMLAttributes<HTMLInputElement>;

export const renderInput: FieldRenderFunction<InputValue, InputProps> = ({ input: { name, value, onChange }, custom }) => (
    <input {...custom} name={name} value={value} onChange={event => onChange(event.target.value)}/>
);

export const InputField = createField<InputValue, InputProps>(renderInput);
