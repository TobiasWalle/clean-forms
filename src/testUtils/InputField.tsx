import * as React from 'react';
import { FieldRenderFunction } from '../lib/components/Field';
import { createField } from '../lib/hocs/index';

export interface InputFieldProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  showErrorIfDirty?: boolean;
}

export const renderInput: FieldRenderFunction<string | number, InputFieldProps> = ({
  input: {
    name,
    value,
    onFocus,
    onChange,
    onBlur,
    error,
    touched,
    dirty
  },
  custom: {
    showErrorIfDirty,
    ...custom
  }
}) => {
  const showError = error && (showErrorIfDirty ? dirty : touched);
  return (
    <label>
      {custom.label}
      <input
        name={name}
        value={value}
        onChange={event => {
          let newValue: string | number = event.target.value;
          if (custom.type === 'number') {
            newValue = Number(newValue);
          }
          onChange(newValue);
        }}
        onFocus={onFocus}
        onBlur={onBlur}
        {...custom}
      />
      {showError && <div>{error}</div>}
    </label>
  );
};

export const InputField = createField<string | number, InputFieldProps>(renderInput);
