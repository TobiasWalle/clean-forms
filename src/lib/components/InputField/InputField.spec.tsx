import { shallow } from 'enzyme';
import * as React from 'react';
import { InputField } from './InputField';

describe('InputField', () => {
  it('should render', () => {
    expect(shallow(<InputField name={'name'}/>)).toMatchSnapshot();
  });
});
