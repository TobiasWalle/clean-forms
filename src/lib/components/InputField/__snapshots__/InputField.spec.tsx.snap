// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render 1`] = `
<Field
  name="name"
  render={[Function]}
/>
`;
